cardano-api/gen/Test/Gen/Cardano/Api/Typed.hs

```diff
@@ -862,6 +863,39 @@ genProtocolParameters =
     <*> Gen.maybe genNat
     <*> Gen.maybe genLovelace
 
+-- | Generate valid protocol parameters which pass validations in Cardano.Api.ProtocolParameters
+genValidProtocolParameters :: Gen ProtocolParameters
+genValidProtocolParameters =
+  ProtocolParameters
+    <$> ((,) <$> genNat <*> genNat)
+    <*> Gen.maybe genRational
+    <*> genMaybePraosNonce
+    <*> genNat
+    <*> genNat
+    <*> genNat
+    <*> genLovelace
+    <*> genLovelace
+    <*> Gen.maybe genLovelace
+    <*> genLovelace
+    <*> genLovelace
+    <*> genLovelace
+    <*> genEpochNo
+    <*> genNat
+    <*> genRationalInt64
+    <*> genRational
+    <*> genRational
+    <*> fmap Just genLovelace
+    <*> return mempty
+    --TODO: Babbage figure out how to deal with
+    -- asymmetric cost model JSON instances
+    <*> fmap Just genExecutionUnitPrices
+    <*> fmap Just genExecutionUnits
+    <*> fmap Just genExecutionUnits
+    <*> fmap Just genNat
+    <*> fmap Just genNat
+    <*> fmap Just genNat
+    <*> fmap Just genLovelace
```

Comment:
What's the reason for using `Just`?

Reply:
When generating `TxBody` , it is validated using functions like: https://github.com/input-output-hk/cardano-node/blob/a75c331a02cfd1f19f45dbb21a89728cb73b4f6a/cardano-api/src/Cardano/Api/ProtocolParameters.hs#L1338 which checks for present protocol parameters. If they are not there, we're getting the error instead of `TxBody`.

Reply:
Is this true of all the fields that use `Just`?  Can you add a comment to that effect?

Reply:
Yes, I'll add a comment.